x-pack/test/alerting_api_integration/common/plugins/alerts/server/action_types.ts

```diff
@@ -139,6 +140,96 @@ function getIndexRecordActionType() {
   return result;
 }
 
+function getHookedActionType() {
+  const paramsSchema = schema.object({});
+  type ParamsType = TypeOf<typeof paramsSchema>;
+  const configSchema = schema.object({
+    index: schema.string(),
+    source: schema.string(),
+  });
+  type ConfigType = TypeOf<typeof configSchema>;
+  const secretsSchema = schema.object({
+    encrypted: schema.string(),
+  });
+  type SecretsType = TypeOf<typeof secretsSchema>;
+  const result: ActionType<ConfigType, SecretsType, ParamsType> = {
+    id: 'test.connector-with-hooks',
+    name: 'Test: Connector with hooks',
+    minimumLicenseRequired: 'gold',
+    supportedFeatureIds: ['alerting'],
+    validate: {
+      params: { schema: paramsSchema },
+      config: { schema: configSchema },
+      secrets: { schema: secretsSchema },
+    },
+    async executor({ config, secrets, params, services, actionId }) {
+      return { status: 'ok', actionId };
+    },
+    async preSaveHook({ connectorId, config, secrets, services, isUpdate, logger }) {
```

Comment:
nit: could add a flag to the config to force the hook to throw an error to test the error condition

Reply:
ah, thanks for reminding me.  I do think we should have some FT that test the error conditions, but didn't think they were absolutely required, since we do test in the jest tests.  And given the time frame.  Also not sure how to test for the SO create/update failing in FT, but could likely do it in a jest integration test.  

So was planning on opening an issue to add those later.